### DISCUSSION TITLE
When Clicking SUBMIT Have An Are You Sure? Pop-Up

### DISCUSSION THREAD

**Discussion by wellonthehill** [codewars/codewars.com#2799] [Ideas]
Sometimes I don't realize the Attempt button has become a Submit button and i accidentally click it. When it becomes the Submit button it would be nice if a popup asked - 'Are you sure you wish to submit?'

**Comment by hobovsky** [comment]
Why would you need it? What exactly is wrong with submitting your solution that you might want to avoid it?

**Comment by Kacarott** [comment]
This seems far more annoying than useful. A better way to solve this issue might be to request the ability for solvers to hide their own solutions.

**Comment by Blind4Basics** [comment]
> A better way to solve this issue might be to request the ability for solvers to hide their own solutions.

'won't happen. ;p

**Comment by JohanWiltink** [comment]
Don't _click_ `Submit / Attempt` - use the keyboard. `Ctrl-;` does `Attempt` but not `Submit`. ( `Ctrl-S` or `Ctrl-'` does `Test`; `Ctrl-Enter` does `Attempt` _or_ `Submit`. )

( This is on Windows and, from memory, Linux. Mac has key combos as well, but different ones. )

**Comment by wellonthehill** [comment]
you might be working on your code for 15 to 20 minutes. clicking one verify popup hardly seems to be an inconvenience.  good GUI design would not let a serious commit act occur on the same button as a button you use over and over while testing things. when the button changes to submit it is a subtle change.  Safe GUI design would have either a 'confirm' popup or a third button that was non 'attempt.'  The safer practice is a confirm popup since there is no undo after an accidental submit

    On Monday, September 5, 2022 at 01:25:57 PM EDT, Keldan Chapman ***@***.***> wrote:  
 
 


This seems far more annoying than useful. A better way to solve this issue might be to request the ability for solvers to hide their own solutions.

**Reply by hobovsky** [reply]
> good GUI design would not let a serious commit act occur on the same button as a button you use over and over while testing things.

Is submitting a solution really a "serious commit act"?

**Comment by wellonthehill** [comment]
> On Monday, September 5, 2022 at 02:33:20 PM EDT, JohanWiltink ***@***.***> wrote:  
> Don't click Submit / Attempt - use the keyboard. Ctrl-; does Attempt but not Submit. ( Ctrl-S or Ctrl-' does Test. )

thanks for the keyboard tip 

> On Monday, September 5, 2022 at 03:48:09 AM EDT, hobovsky ***@***.***> wrote:  
 > Why would you need it? What exactly is wrong with submitting your solution that you might want to avoid it?

sometimes a console.debug statement is still there OR i haven't refactored enough

> On Monday, September 5, 2022 at 09:03:45 PM EDT, hobovsky ***@***.***> wrote:  
> Is submitting a solution really a "serious commit act"?

in the context of that system it is since there is no undo or post-editing after submit (as far as i know, but hey i'm new)


Paul
Kindness in words creates confidence. Kindness in thinking creates profoundness. Kindness in giving creates love.Lao Tzu

**Reply by Blind4Basics** [reply]
@wellonthehill : please don't reply in separate messages, especially using mail reply format, which is becoming a total mess on github. I had to edit and merge your three messages to make everything readable again.

Thx & cheers
